pkg/acsengine/engine.go

```diff
@@ -658,6 +657,15 @@ func getParameters(cs *api.ContainerService, isClassicMode bool, generatorCode s
 			addValue(parametersMap, "aadTenantId", properties.AADProfile.TenantID)
 			addValue(parametersMap, "aadServerAppId", properties.AADProfile.ServerAppID)
 		}
+
+		if properties.OrchestratorProfile.KubernetesConfig.KubeletConfig != nil {
+			for key, val := range properties.OrchestratorProfile.KubernetesConfig.KubeletConfig {
+				switch key {
+				case "--cluster-dns":
```

Comment:
Out of curiosity, what's the reasoning behind naming this "--cluster-dns" vs "cluster-dns" since in this context it's not used as an arg but as a json string?

Reply:
Good question. The thinking is to transparently use the exact string that we'll be passing to the component at runtime, so that we may eventually just pass it on as-is. Also, `--` as a prefix might not be generally valid going forward, it's possible some options might have a `-` prefix (or some other prefix).